cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -709,6 +716,37 @@ protected Future<Void> certificateSecret(Clock clock) {
                 });
     }
 
+    /**
+     * Emits the certificate expiration metric for the given secret if the resultType is CREATED or PATCHED,
+     * if DELETED the metric will be set to 0 for resetting.
+     *
+     * @param resultType Type of the ReconcileResult
+     * @param secret The modified secret
+     * @param crtKey The crt key for the modified secret
+     */
+    private void emitCertificateSecretMetrics(ReconcileResult.Type resultType, Secret secret, String crtKey) {
```

Comment:
Sorry, but I don't follow this logic. Why does the result type matter? The CA will always exist.

Reply:
@scholzj The idea of the type was to not do instanceof checks on the class like:
```
if (patchResult instanceof ReconcileResult.Patched) {
   ...
}
```

With regards to you comment about the CA always exists? I guess DELETED will also be called and in that case we need to remove the metrics, which is another case than CREATED and PATCHED.

Reply:
I think that this is badly placed:
* The `KafkaRecocniler.certificateSecret()` method manages only the secret (secrets in the future if we one day implement 7687) with the broker certificates. Right now, this secret will never be deleted in this part of the code. It will be deleted only through Kubernetes garbage collection. So I don't think the `DELETED` case will ever happen. And even if it did, it says absolutely nothing about the CA. 
* The secret created in `KafkaRecocniler.certificateSecret()` will have entry per broker. And right now, the code seems to set the metric for each broker based on the broker certificate expiration which is not the same as the CA expiration IIRC.  Yet, you seem to use the same metric, so you IMHO just override one value again and again.

My understanding is that we want to focus right now on the CAs, or? Not on the server certificates? in that case, maybe you can place it into one of these two places:
* The CAs are managed in the CaReconciler class. So I think the metric could be emitted there? 
* Alternatively, the `ReconciliationState` inside the `KafkaAssemblyOperator` class holds both the Cluster and Clients CA which should be able to provide the expiration. Maybe that is an even better place?
 
In any case, if we want the CAs, I think those places will be better suited to emit he metric than the KafkaRecocniler.

_(Hopefully I remembered this all correctly what the code does)_

Reply:
Right, showing cluster and user CAs is probably better.